forte/data/readers/tests/html_reader_test.py

```diff
@@ -0,0 +1,209 @@
+import os
+import shutil
+import unittest
+from pathlib import Path
+from ddt import ddt, data
+
+from forte.pipeline import Pipeline
+from forte.data.base import Span
+from forte.data.data_pack import DataPack
+from forte.data.readers import HTMLReader
+
+
+@ddt
+class HTMLReaderPipelineTest(unittest.TestCase):
+    def setUp(self):
+        self._cache_directory = Path(os.path.join(os.getcwd(), "cache_html"))
+        self.reader = HTMLReader(cache_directory=self._cache_directory,
+                                 append_to_cache=True)
+
+        self.pl1 = Pipeline()
+        self.pl1.set_reader(self.reader)
+
+        self.pl2 = Pipeline()
+        self.pl2.set_reader(HTMLReader(from_cache=True,
+                                       cache_directory=self._cache_directory))
+
+        self.data_dir = os.path.join(
```

Comment:
Initialize the pipeline after creation. `self.pl1.initialize()` and `self.pl2.initialize()`